main.py

```diff
@@ -92,71 +97,79 @@ def _get_link_from_interface(self, interface):
-                error_msg = (f"Error restoring link endpoint: {error}")
-                raise KeyError(error_msg)
-
-            link = self._get_link_or_create(endpoint_a, endpoint_b)
-            endpoint_a.update_link(link)
-            endpoint_b.update_link(link)
-
-            endpoint_a.nni = True
-            endpoint_b.nni = True
+    def _restore_link(self, link_id):
+        """Restore administrative state the link saved in storehouse."""
+        try:
+            state = self.links_state[link_id]
+        except KeyError:
+            error = ('The stored topology is different from the current'
+                     f' topology. The link {link_id} has no stored '
+                     'administrative state to restore.')
+            raise KeyError(error)
 
+        try:
+            link = self.links[link_id]
             if state['enabled']:
                 link.enable()
             else:
                 link.disable()
+        except KeyError:
+            error = ('Error restoring link status.'
+                     f'The link {link_id} does not exist.')
+            raise KeyError(error)
+        log.info(f'The state of link {link.id} has been restored.')
+        self.notify_topology_update()
+        self.update_instance_metadata(link)
 
-            self.notify_topology_update()
-            self.update_instance_metadata(link)
-            log.info(f"Link {link.id} retrieved.")
-
-    def _restore_status(self):
-        """Restore the network administrative status saved in StoreHouse."""
-        # restore Switches
-        for switch_id, state in self.switches_state.items():
-            try:
-                if state:
-                    self.controller.switches[switch_id].enable()
-                else:
-                    self.controller.switches[switch_id].disable()
-            except KeyError:
-                error = ('Error while restoring switches status. The '
-                         f'{switch_id} does not exist.')
-                raise KeyError(error)
+    def _restore_switch(self, switch_id):
+        """Restore administrative state the switches saved in storehouse."""
+        # restore switches
+        try:
+            state = self.switches_state[switch_id]
+        except KeyError:
+            error = ('The stored topology is different from the current'
+                     f' topology. The switch {switch_id} has no stored'
+                     ' administrative state to restore.')
+            raise KeyError(error)
+        try:
+            switch = self.controller.switches[switch_id]
+            if state:
+                switch.enable()
+            else:
+                switch.disable()
+        except KeyError:
+            error = ('Error while restoring switches status. The '
+                     f'{switch_id} does not exist.')
+            raise KeyError(error)
+        # wait to restore interfaces
+        while not switch.interfaces:
+            time.sleep(self.interface_restore)
```

Comment:
I think we should have a log.debug here, so we can detect if this code gets stuck for some reason.